modules/gapi/test/infer/gapi_infer_ie_test.cpp

```diff
@@ -75,6 +55,63 @@ void normAssert(cv::InputArray ref, cv::InputArray test,
     EXPECT_LE(normInf, lInf) << comment;
 }
 
+namespace IE = InferenceEngine;
+
+void setNetParameters(IE::CNNNetwork& net) {
+    auto &ii = net.getInputsInfo().at("data");
+    ii->setPrecision(IE::Precision::U8);
+    ii->setLayout(IE::Layout::NHWC);
+    ii->getPreProcess().setResizeAlgorithm(IE::RESIZE_BILINEAR);
+}
+
+#if INF_ENGINE_RELEASE < 2020000000  // < 2020.1
+IE::CNNNetwork readNetwork(const std::string& topology_path,
+                           const std::string& weights_path) {
+    IE::CNNNetReader reader;
+    reader.ReadNetwork(topology_path);
+    reader.ReadWeights(weights_path);
+    return reader.getNetwork();
+}
+
+IE::CNNNetwork readNetwork(IE::Core&,
+                           const std::string& topology_path,
+                           const std::string& weights_path) {
+    return readNetwork(topology_path, weights_path);
+}
+#else // >= 2020.1
+inline IE::CNNNetwork readNetwork(      IE::Core&    core,
+                                  const std::string& topology_path,
+                                  const std::string& weights_path) {
+    return core.ReadNetwork(topology_path, weights_path);
+}
+#endif // INF_ENGINE_RELEASE < 2020000000
+#if INF_ENGINE_RELEASE < 2019020000  // < 2019.R2
+inline IE::InferencePlugin getPlugin(const std::string& device) {
+    return IE::PluginDispatcher().getPluginByDevice(device);
+}
+
+IE::CNNNetwork readNetwork(IE::InferencePlugin&,
+                           const std::string& topology_path,
+                           const std::string& weights_path) {
+    return readNetwork(topology_path, weights_path);
+}
+
+inline IE::ExecutableNetwork loadNetwork(      IE::InferencePlugin& plugin,
+                                         const IE::CNNNetwork&      net,
+                                         const std::string&) {
+    return plugin.LoadNetwork(net, {});
+}
+#else // >= 2019.R2
+inline IE::Core getPlugin(const std::string&) {
+    return IE::Core();
+}
+
+inline IE::ExecutableNetwork loadNetwork(      IE::Core&       core,
+                                         const IE::CNNNetwork& net,
+                                         const std::string&    device) {
+    return core.LoadNetwork(net, device);
+}
+#endif // INF_ENGINE_RELEASE < 2019020000
```

Comment:
To avoid this duplication, maybe you can put the whole stuff to a header (under `giebackend/`) and reuse this header here.

Our tests can refer to our internal headers.

Reply:
created a "giebackend/gieapi.hpp" header